packages/ns-api/files/ns.threatshield

```diff
@@ -541,6 +542,141 @@ def dns_delete_bypass(e_uci, payload):
+                "val": __parse_expression(expression["elem"]["val"]),
+            }
+
+    raise NotImplementedError('Unsupported expression type: {}'.format(type(expression)))
+
+
+def __parse_set(blocklist):
+    """
+    This function parses any set returned by nftables and returns a list of entries
+    """
+    blocked_ips = []
+    if ("nftables" in blocklist
+            and len(blocklist["nftables"]) > 1
+            and "set" in blocklist["nftables"][1]
+            and "elem" in blocklist["nftables"][1]["set"]):
+        blocked_ips = __parse_expression(blocklist["nftables"][1]["set"]["elem"])
+
+    return blocked_ips
+
+
+def list_active_blocks():
+    try:
+        output_ipv4 = subprocess.run(["nft", "-j", "list", "set", "inet", "banIP", "blocklistv4"], check=True,
+                                    capture_output=True, text=True)
+        data = json.loads(output_ipv4.stdout)
+    except subprocess.CalledProcessError:
+        data = {}
+
+    blocks = __parse_set(data)
+
+    try:
+        output_ipv6 = subprocess.run(["nft", "-j", "list", "set", "inet", "banIP", "blocklistv6"], check=True,
+                                    capture_output=True, text=True)
+        data = json.loads(output_ipv6.stdout)
+    except subprocess.CalledProcessError:
+        data = {}
+
+    blocks = blocks + __parse_set(data)
+    blocks = filter(lambda x: isinstance(x, dict) and "val" in x, blocks)
+    blocks = sorted(blocks, key=lambda x: x["timeout"] - x["expires"])
+    response = []
+    for block in blocks:
+        if not isinstance(block["val"], list):
+            block["val"] = [block["val"]]
+        for item in block["val"]:
+            if isinstance(item, ipaddress.IPv4Address) or isinstance(item, ipaddress.IPv6Address):
+                response.append({
+                    "ip": str(item),
+                    "timeout": block["timeout"],
+                    "expires": block["expires"],
+                })
+            if isinstance(item, ipaddress.IPv4Network) or isinstance(item, ipaddress.IPv6Network):
+                for addr in item:
+                    if isinstance(addr, ipaddress.IPv4Address) or isinstance(addr, ipaddress.IPv6Address):
+                        response.append({
+                            "ip": str(addr),
+                            "timeout": block["timeout"],
+                            "expires": block["expires"],
+                        })
+
+    return response
+
+
+def remove_active_block():
+    data = json.loads(sys.stdin.read())
+    if "ip" not in data:
+        return validation_error("ip", "required")
+    if data["ip"] == "":
+        return validation_error("ip", "required")
+
+    try:
+        ip = ipaddress.ip_address(data["ip"])
+    except ValueError:
+        return validation_error("ip", "invalid", data["ip"])
+
+    found = False
+    blocks = filter(lambda x: x["ip"] == str(ip), list_active_blocks())
+    for _ in blocks:
+        found = True
```

Comment:
This is very hard to read.
Can we use something like `found = len(blocks) > 0`?

Reply:
This appears to be the only way, bit ugly but still more readable

```suggestion
    found = len(list(blocks)) > 0
```

Reply:
If possible, I prefer you suggestion which uses len